### ISSUE TITLE
Beatmap download blocked by background import processes

### ISSUE THREAD

**Issue by cupppu** [ppy/osu#9781]
**Describe the bug:** 
as titled, beatmap download (through osu!direct) should have priority over background beatmap or other import processes

**Screenshots or videos showing encountered issue:** 
<img width="302" alt="image" src="https://user-images.githubusercontent.com/7279624/89407877-22d69600-d752-11ea-949f-35471e84e64e.png">
(the second "beatmap import is initialising" is the beatmap download)

**osu!lazer version:** 2020.801.0

**Logs:**
[performance-audio.log](https://github.com/ppy/osu/files/5027995/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/5027996/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/5027997/performance-update.log)
[performance.log](https://github.com/ppy/osu/files/5027998/performance.log)
[runtime.log](https://github.com/ppy/osu/files/5027999/runtime.log)

**Comment by frenzibyte** [comment]
Duplicate of https://github.com/ppy/osu/issues/6322, please search before reporting.
